### PR TITLE
Fix common redundancies by resharper

### PR DESCRIPTION
**Pull request by dampir**
I've removed some common redundancies of code from Octokit.Net.

List of redundancies:

1. Redundant explicit array creation in argument of 'params' parameter.
2. Redundant type qualifiers.
3. Empty constructors.
4.  Redundant explicit array type specification.
5. Redundant member initializers.
6.  Redundant collection list initializers.
7.  Unused "using" directives were removed (in modified files).
8. Some misspellings were fixed.

This PR was made in order to support code quality in Octokit.Net.

### CODE REVIEW

**Comment by Haacked** [comment]
In general, each of these should be individual pull requests. However, in this case, the PR is small enough I'll allow it. :smile: I'll let @shiftkey take a look and merge it.

**Comment by dampir** [comment]
Yep, I thought about separate PR for each type of changes, so I'm going to create small PR in further, because all changes are small and quite simple. There are some common code cleanups that I want to provide.

File: Octokit.Tests.Integration/Clients/OrganizationMembersClientTests.cs [hunk 1]
```diff
@@ -47,7 +42,7 @@ public TheGetAllMethod()
 
                 // There shouldnt be any members that are in both groups if the role filter works correctly
                 var membersInBoth = adminMembers.Select(a => a.Id).Intersect(normalMembers.Select(n => n.Id));
-                Assert.True(membersInBoth.Count() == 0);
+                Assert.True(!membersInBoth.Any());
```
Comment (shiftkey) [inline Octokit.Tests.Integration/Clients/OrganizationMembersClientTests.cs]: :lipstick: let's make this `Assert.NotEmpty(membersInBoth);`

Comment (dampir) [inline Octokit.Tests.Integration/Clients/OrganizationMembersClientTests.cs]: Done!

**Comment by devkhan** [comment]
@dampir Can you make the list in the first comment a to-do list? Or if everything is done, ignore this.

**Comment by dampir** [comment]
@devkhan, everything is done.

**Comment by shiftkey** [comment]
I have some reservations about the `SimpleJson` changes, mostly due to that file being an external dependency we pull in. Are you able to revert that change?

**Comment by Haacked** [comment]
> I have some reservations about the SimpleJson changes, mostly due to that file being an external dependency we pull in. Are you able to revert that change?

Good catch! Yes, we don't want to change that file. That breaks NuGet's ability to detect if there are any changes or not when upgrading the package.

**Comment by dampir** [comment]
Changes has been reverted. I'll exclude that file from Resharer's analysis.

File: Octokit.Tests.Integration/Clients/OrganizationMembersClientTests.cs [hunk 2]
```diff
@@ -47,7 +42,7 @@ public TheGetAllMethod()
 
                 // There shouldnt be any members that are in both groups if the role filter works correctly
                 var membersInBoth = adminMembers.Select(a => a.Id).Intersect(normalMembers.Select(n => n.Id));
-                Assert.True(membersInBoth.Count() == 0);
+                Assert.NotEmpty(membersInBoth);
```
Comment (shiftkey) [inline Octokit.Tests.Integration/Clients/OrganizationMembersClientTests.cs]: Apologies: I inverted this assert earlier - this should be `Assert.Empty`

Comment (dampir) [inline Octokit.Tests.Integration/Clients/OrganizationMembersClientTests.cs]: Done!

**Comment by shiftkey** [comment]
@dampir excellent, thanks!

**Pull request merged**